### PR TITLE
placementinsql: add the survival preferences docs

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/12682
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: placement-rules-in-sql.md [hunk 1]
```diff
@@ -135,6 +135,7 @@ In addition to the placement options above, you can also use the advance configu
 | `FOLLOWER_CONSTRAINTS`     | A list of constraints that only apply to followers.                                   |
 | `LEARNER_CONSTRAINTS`      | A list of constraints that only apply to learners.                                     |
 | `LEARNERS`                 | The number of learners. |
+| `SURVIVAL_PREFERENCE`      | The replica placement priority according to the label of the disaster tolerance level. For example, `SURVIVAL_PREFERENCE="[region, zone, host]"`. |
```
Comment (qiancai) [inline placement-rules-in-sql.md:138]:
```suggestion
| `SURVIVAL_PREFERENCE`      | Specifies the replica placement priority according to the disaster tolerance level of the labels. For example, `SURVIVAL_PREFERENCE="[region, zone, host]"`. |
```

File: placement-rules-in-sql.md [hunk 2]
```diff
@@ -248,6 +249,25 @@ In dictionary format, constraints also indicate a number of instances that apply
...
+For example, you can set the `SURVIVAL_PREFERENCES` as follows to require the data to meet the configured survival preferences as much as possible:
+
+``` sql
+CREATE PLACEMENT POLICY multiregion
+    follower=4
+    PRIMARY_REGION="region1"
+    SURVIVAL_PREFERENCES="[region, zone]";
+```
+
+Then, for tables with this policy, data will first meet the survival objectives of data isolation across regions, and then ensure the survival objectives of data isolation across availability zones.
```
Comment (qiancai) [inline placement-rules-in-sql.md:265]:
```suggestion
Then, for tables attached with this policy, data will first meet the survival objectives of data isolation across regions, and then meet the survival objectives of data isolation across availability zones.
```

File: placement-rules-in-sql.md [hunk 3]
```diff
@@ -248,6 +249,25 @@ In dictionary format, constraints also indicate a number of instances that apply
 >
 > Dictionary and list formats are based on the YAML parser, but the YAML syntax might be incorrectly parsed. For example, `"{+disk=ssd:1,+disk=nvme:2}"` is incorrectly parsed as `'{"+disk=ssd:1": null, "+disk=nvme:1": null}'`. But `"{+disk=ssd: 1,+disk=nvme: 1}"` is correctly parsed as `'{"+disk=ssd": 1, "+disk=nvme": 1}'`.
 
+### Survival Preferences
+
+For some important data, you might want to have multiple replicas across availability zones to get high disaster survivability, such as survivability at the cloud region level. To achieve that, you can set `SURVIVAL_PREFERENCES` for your data.
+
+For example, you can set the `SURVIVAL_PREFERENCES` as follows to require the data to meet the configured survival preferences as much as possible:
```
Comment (qiancai) [inline placement-rules-in-sql.md:256]:
```suggestion
For example, when creating a placement policy, you can set the `SURVIVAL_PREFERENCES` as follows to require the data to meet the configured survival preferences as much as possible:
```

File: placement-rules-in-sql.md [hunk 4]
```diff
@@ -248,6 +249,25 @@ In dictionary format, constraints also indicate a number of instances that apply
 >
 > Dictionary and list formats are based on the YAML parser, but the YAML syntax might be incorrectly parsed. For example, `"{+disk=ssd:1,+disk=nvme:2}"` is incorrectly parsed as `'{"+disk=ssd:1": null, "+disk=nvme:1": null}'`. But `"{+disk=ssd: 1,+disk=nvme: 1}"` is correctly parsed as `'{"+disk=ssd": 1, "+disk=nvme": 1}'`.
 
+### Survival Preferences
```
Comment (qiancai) [inline placement-rules-in-sql.md:252]:
```suggestion
### Survival preferences
```

**Review by nolouch — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

File: placement-rules-in-sql.md [hunk 5]
```diff
@@ -248,6 +249,25 @@ In dictionary format, constraints also indicate a number of instances that apply
...
+### Survival preferences
+
+For some important data, you might want to store multiple replicas across availability zones to get high disaster survivability, such as survivability at the cloud region level. When you create or modify a placement policy, you can use `SURVIVAL_PREFERENCES` to set the preferred survivability for your data.
+
+For example, when creating a placement policy, you can set the `SURVIVAL_PREFERENCES` as follows to require the data to meet the configured survival preferences as much as possible:
+
+``` sql
+CREATE PLACEMENT POLICY multiregion
+    follower=4
+    PRIMARY_REGION="region1"
```
Comment (hfxsd) [inline placement-rules-in-sql.md:261]:
```suggestion
    PRIMARY_REGION="us-east-1"
```

File: placement-rules-in-sql.md [hunk 6]
```diff
@@ -248,6 +249,25 @@ In dictionary format, constraints also indicate a number of instances that apply
...
+    SURVIVAL_PREFERENCES="[region, zone]";
+```
+
+Then, for tables attached with this policy, data will first meet the survival objectives of data isolation across regions, and then meet the survival objectives of data isolation across availability zones.
```
Comment (hfxsd) [inline placement-rules-in-sql.md:265]:
```suggestion
Then, for the tables attached with this policy, the data will first meet the survival objectives of data isolation across regions, and then meet the survival objectives of data isolation across availability zones.
```

File: placement-rules-in-sql.md [hunk 7]
```diff
@@ -248,6 +249,25 @@ In dictionary format, constraints also indicate a number of instances that apply
...
+
+> **Note:**
+>
+> `SURVIVAL_PREFERENCES` is equivalent to `LOCATION_LABELS` in PD. For more information, see [Schedule replicas by topology labels](/schedule-replicas-by-topology-labels.md).
```
Comment (hfxsd) [inline placement-rules-in-sql.md:269]:
```suggestion
> `SURVIVAL_PREFERENCES` is equivalent to `LOCATION_LABELS` in PD. For more information, see [Schedule Replicas by Topology Labels](/schedule-replicas-by-topology-labels.md).
```

**Review by hfxsd — APPROVED** [review]
Rest LGTM

File: placement-rules-in-sql.md [hunk 8]
```diff
@@ -135,6 +135,7 @@ In addition to the placement options above, you can also use the advance configu
 | `FOLLOWER_CONSTRAINTS`     | A list of constraints that only apply to followers.                                   |
 | `LEARNER_CONSTRAINTS`      | A list of constraints that only apply to learners.                                     |
 | `LEARNERS`                 | The number of learners. |
+| `SURVIVAL_PREFERENCE`      | Specifies the replica placement priority according to the disaster tolerance level of the labels. For example, `SURVIVAL_PREFERENCE="[region, zone, host]"`. |
```
Comment (hfxsd) [inline placement-rules-in-sql.md:138]:
```suggestion
| `SURVIVAL_PREFERENCE`      | The replica placement priority according to the disaster tolerance level of the labels. For example, `SURVIVAL_PREFERENCE="[region, zone, host]"`. |
```

File: placement-rules-in-sql.md [hunk 9]
```diff
@@ -248,6 +249,26 @@ In dictionary format, constraints also indicate a number of instances that apply
...
+For example, assuming that you have a TiDB cluster across 3 availability zones, with multiple TiKV instances deployed on each host in each zone. And when creating placement policies for this cluster, you have set the `SURVIVAL_PREFERENCES` as follows:
+
+``` sql
+CREATE PLACEMENT POLICY multiaz SURVIVAL_PREFERENCES="[zone, host]";
+CREATE PLACEMENT POLICY singleaz CONSTRAINTS="[+zone=zone1]" SURVIVAL_PREFERENCES="[host]";
+```
+
+After creating the placement policies, you can attach them to the corresponding tables as needed:
+
+- For tables attached with the `multiaz` placement policy, data will be placed in 3 replicas in different availability zones, prioritizing survival goals of data isolation cross zones, followed by survival goals of data isolation cross hosts.
```
Comment (hfxsd) [inline placement-rules-in-sql.md:265]: 原文是 绑定，用 attached 还是用 bound?

Comment (qiancai) [reply placement-rules-in-sql.md:265]: 这篇中文文档里提到了多次“绑定”，对应的英文里统一用了 attach （https://docs.pingcap.com/tidb/dev/placement-rules-in-sql），所以这里用了 attach

**Pull request merged**